core/src/deserialize.rs

```diff
@@ -30,6 +35,70 @@ macro_rules! deserialize_number {
     };
 }
 
+#[derive(Debug, PartialEq)]
+pub enum EvalOrDeserError {
+    Nickel(crate::error::Error),
```

Comment:
Maybe a `Nickel { error: crate::error::Error, files: crate::files::Files }` would be better here, since it would allow the caller to print nice diagnostics if they want to.

Reply:
I'm having trouble figuring out the right way to turn an `Input` into a `Files`. When constructing `EvalOrDeserError` from a `crate::error::Error`, what should `files` look like? It seems like there is a bit of a chicken and egg problem, where constructing `Files` requires reading a file, which can fail, which can return the very error we're trying to construct...